packages/migrate/src/commands/DbExecute.ts

```diff
@@ -0,0 +1,171 @@
+
+  ${chalk.dim('$')} prisma db execute --preview-feature [options]
+
+${chalk.bold('Options')}
+
+-h, --help   Display this help message
+     --url   URL of the datasource to run the command on
+  --schema   Path to your Prisma schema file to take the datasource URL from
+   --stdin   Use the terminal standard input as the script to be executed.
+    --file   Path to a file. The content will be sent as the script to be executed.
+
+${chalk.bold('Examples')}
+ 
+  Execute the content of a SQL script file to the database URL from schema.
+  ${chalk.dim('$')} prisma db execute --preview-feature --file ./script.sql --schema schema.prisma
+
+  Execute the SQL script from stdin to the database url.
+  ${chalk.dim(
+    '$',
+  )} echo 'SELECT \`hello world\`;' | prisma db execute --preview-feature --stdin --url="mysql://localhost/testdb"
+
+`)
+
+  public async parse(argv: string[]): Promise<string | Error> {
+    const args = arg(
+      argv,
+      {
+        '--help': Boolean,
+        '-h': '--help',
+        '--stdin': Boolean,
+        '--file': String,
+        '--schema': String,
+        '--url': String,
+        '--preview-feature': Boolean,
+        '--telemetry-information': String,
+      },
+      false,
+    )
+
+    if (isError(args)) {
+      return this.help(args.message)
+    }
+
+    if (args['--help']) {
+      return this.help()
+    }
+
+    if (!args['--preview-feature']) {
+      throw new DbExecuteNeedsPreviewFeatureFlagError()
+    }
+
+    // One of --stdin or --file is required
+    if (args['--stdin'] && args['--file']) {
+      throw new Error(
+        `--stdin and --file cannot be used at the same time. Only 1 must be provided. 
+See \`${chalk.green(getCommandWithExecutor('prisma db execute -h'))}\``,
+      )
+    } else if (!args['--stdin'] && !args['--file']) {
+      throw new Error(
+        `Either --stdin or --file must be provided.
+See \`${chalk.green(getCommandWithExecutor('prisma db execute -h'))}\``,
+      )
+    }
+
+    // One of --url or --schema is required
+    if (args['--url'] && args['--schema']) {
+      throw new Error(
+        `--url and --schema cannot be used at the same time. Only 1 must be provided.
+See \`${chalk.green(getCommandWithExecutor('prisma db execute -h'))}\``,
+      )
+    } else if (!args['--url'] && !args['--schema']) {
+      throw new Error(
+        `Either --url or --schema must be provided.
+See \`${chalk.green(getCommandWithExecutor('prisma db execute -h'))}\``,
+      )
+    }
+
+    let script = ''
+    // Read file
+    if (args['--file']) {
+      script = fs.readFileSync(path.resolve(args['--file']), 'utf-8')
+    }
+    // Read stdin
+    if (args['--stdin']) {
+      script = await getStdin()
+      // If input is empty, stop here
+      if (!script) {
+        throw new Error(
+          `--stdin was passed but the input was empty. See \`${chalk.green(
+            getCommandWithExecutor('prisma db execute -h'),
+          )}\``,
+        )
```

Comment:
I'll remove

Reply:
That's an interesting question. Since we only report back that the command ran successfully users might not be aware if they messed up the stdin passing if we don't fail here. 
On the other hand there is nothing wrong with empty input and other shell commands wouldn't fail in that situation either I think.

Reply:
I also expect most databases will return an error on empty input

Reply:
Actually it only fails on MySQL, see newest commit with tests named `... with empty --file --schema`
https://github.com/prisma/prisma/pull/11371/commits/aec52d434da92c14d9b80f0e1ecaf79d3f7c4d33